### PR TITLE
Make all test use OPERATOR_TEST_STORAGE_CLASS env variable

### PR DESCRIPTION
**Pull request by rohitsakala**
[#166924183](https://www.pivotaltracker.com/n/projects/2192232/stories/166924183)

### CODE REVIEW

File: docs/examples/extended-statefulset/exstatefulset_pvcs.yaml [hunk 1]
```diff
@@ -30,4 +30,4 @@ spec:
           accessModes: [ "ReadWriteOnce" ]
           resources:
             requests:
-              storage: 1Gi              
+              storage: 1Gi
```
Comment (edwardstudy) [inline docs/examples/extended-statefulset/exstatefulset_pvcs.yaml]: newline

File: e2e/kube/examples_test.go [hunk 2]
```diff
@@ -159,9 +189,17 @@ var _ = Describe("Examples", func() {
 			It("bosh-deployment with a persistent disk example must work", func() {
 				yamlFilePath := examplesDir + "bosh-deployment/boshdeployment-with-persistent-disk.yaml"
 
+				By("Creating a secret for implicit variable")
```
Comment (edwardstudy) [inline e2e/kube/examples_test.go]: variable?

File: integration/environment/machine.go [hunk 3]
```diff
@@ -212,7 +212,7 @@ func (m *Machine) PVCsDeleted(namespace string) (bool, error) {
 
 // WaitForStatefulSet blocks until all statefulset pods are running. It fails after the timeout.
 func (m *Machine) WaitForStatefulSet(namespace string, labels string) error {
-	return wait.PollImmediate(m.pollInterval, m.pollTimeout, func() (bool, error) {
+	return wait.PollImmediate(m.pollInterval, 500*time.Second, func() (bool, error) {
```
Comment (edwardstudy) [inline integration/environment/machine.go]: Why not increase `m.pollTimeout`? Is this timeout special?

Comment (viovanov) [reply integration/environment/machine.go]: Why wait 8 minutes? This is a bad idea.
We should redesign our tests if something like this is required.

Comment (rohitsakala) [reply integration/environment/machine.go]: I was just testing this flaky test. I added a wip tag too.

File: docs/examples/extended-statefulset/exstatefulset_pvcs.yaml (hunk 1 above)
Comment (rohitsakala) [reply docs/examples/extended-statefulset/exstatefulset_pvcs.yaml]: Done.

File: e2e/kube/examples_test.go (hunk 2 above)
Comment (rohitsakala) [reply e2e/kube/examples_test.go]: The `bosh-deployment-persistent-disk.yaml` example uses an implicit variable to specify the storage class for the persistent disk. For the implicit variable, I am creating a secret to specify the value of the storage class. (https://github.com/cloudfoundry-incubator/cf-operator/blob/rohit/storageclass/docs/examples/bosh-deployment/boshdeployment-with-persistent-disk.yaml#L21)

File: integration/environment/machine.go (hunk 3 above)
Comment (viovanov) [reply integration/environment/machine.go]: Ah, I got scared :P

**Pull request closed without merging**